File: Dddml.Wms.HttpServices.JavaClient/src/org/dddml/wms/domain/OrganizationStructureIdFlattenedDto.java
package org.dddml.wms.domain;

import java.util.HashMap;
import java.util.Map;

public class OrganizationStructureIdFlattenedDto
{


    public static final String[] propertyNames = new String[]{
            "organizationStructureTypeId",
            "parentId",
            "subsidiaryId",
        };

    public static final String[] propertyTypes = new String[]{
            "String",
            "String",
            "String",
        };

    public static final Map<String, String> propertyTypeMap;

    static {
        propertyTypeMap = new HashMap<String, String>();
        for (int i = 0; i < propertyNames.length; i++ ) {
            propertyTypeMap.put(propertyNames[i], propertyTypes[i]);
        }
    }
	
    private OrganizationStructureId value;

    public OrganizationStructureIdFlattenedDto()
    {
        this(new OrganizationStructureId());
    }

    public OrganizationStructureIdFlattenedDto(OrganizationStructureId value)
    {
        this.value = value;
    }

    public OrganizationStructureId toOrganizationStructureId()
    {
        return this.value;
    }


    public String getOrganizationStructureTypeId()
    {
        return this.value.getOrganizationStructureTypeId();
    }

    public void setOrganizationStructureTypeId(String organizationStructureTypeId)
    {
        this.value.setOrganizationStructureTypeId(organizationStructureTypeId);
    }

    public String getParentId()
    {
        return this.value.getParentId();
    }

    public void setParentId(String parentId)
    {
        this.value.setParentId(parentId);
    }

    public String getSubsidiaryId()
    {
        return this.value.getSubsidiaryId();
    }

    public void setSubsidiaryId(String subsidiaryId)
    {
        this.value.setSubsidiaryId(subsidiaryId);
    }


    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        OrganizationStructureIdFlattenedDto other = (OrganizationStructureIdFlattenedDto)obj;
        return value.equals(other.value);
    }

    @Override
    public int hashCode()
    {
        return value.hashCode();
    }

}

File: Dddml.Wms.HttpServices.PhpClient/Wms/Domain/OrganizationStructureIdFlattenedDto.php
<?php

namespace Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\FlattenedDtoInterface;

class OrganizationStructureIdFlattenedDto implements FlattenedDtoInterface
{

    const PROPERTIES = [
            'organizationStructureTypeId' => 'string',
            'parentId' => 'string',
            'subsidiaryId' => 'string',
        ];

    public static function getPropertyNames()
    {
        return array_keys(static::PROPERTIES);
    }

    public static function getPropertyTypes()
    {
        return array_values(static::PROPERTIES);
    }

	
    /**
     * @var OrganizationStructureId
     */
    private $value;

    /**
     * @param OrganizationStructureId $value
     */
    public function __construct(OrganizationStructureId $value = null)
    {
        if ($value) {
            $this->value = $value;
        } else {
            $this->value = new OrganizationStructureId();
        }
    }

    /**
     * @return OrganizationStructureId
     */
    public function toOrganizationStructureId()
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function getOrganizationStructureTypeId()
    {
        return $this->value->getOrganizationStructureTypeId();
    }

    /**
     * @param string $organizationStructureTypeId
     */
    public function setOrganizationStructureTypeId($organizationStructureTypeId)
    {
        $this->value->setOrganizationStructureTypeId($organizationStructureTypeId);
    }

    /**
     * @return string
     */
    public function getParentId()
    {
        return $this->value->getParentId();
    }

    /**
     * @param string $parentId
     */
    public function setParentId($parentId)
    {
        $this->value->setParentId($parentId);
    }

    /**
     * @return string
     */
    public function getSubsidiaryId()
    {
        return $this->value->getSubsidiaryId();
    }

    /**
     * @param string $subsidiaryId
     */
    public function setSubsidiaryId($subsidiaryId)
    {
        $this->value->setSubsidiaryId($subsidiaryId);
    }

    /**
     * @return string
     */
    public function toString()
    {
        $pValues = [
            $this->getOrganizationStructureTypeId(),
            $this->getParentId(),
            $this->getSubsidiaryId(),
        ];
        return implode(',', $pValues);
    }

    /**
     * @param string $data
     *
     * @return OrganizationStructureIdFlattenedDto
     */
    public function fromString($data)
    {
        $pValues = explode(',', $data);
        $this->setOrganizationStructureTypeId($pValues[0]);
        $this->setParentId($pValues[1]);
        $this->setSubsidiaryId($pValues[2]);
        return $this;
    }

}

